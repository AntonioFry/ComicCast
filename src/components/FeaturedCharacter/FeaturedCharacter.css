.featured-character-article {
  border: 1px solid black;
  box-shadow: 0px 0px 10px 1px;
  display: flex;
  height: 250px;
  margin: auto;
  width: 90%;
}

.featured-character-img {
  box-shadow: 0 10px 0px -10px;
  max-width: 250px;
  max-height: 250px;
}

.featured-character-info {
  box-shadow: -5px 1px 10px;
  background-color: black;
  padding: 10px 20px;
  width: 100%;
}

.featured-character-text {
  color: white;
}