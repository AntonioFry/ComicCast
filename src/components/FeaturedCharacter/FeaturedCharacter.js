import React from 'react';
import { connect } from 'react-redux';
import './FeaturedCharacter.css'
import PropTypes from 'prop-types';

export const FeaturedCharacter = (props) => {
  const featuredCharacter = props.allCharacters[1]
  return (
    <article className="featured-character-article">
      <img className="featured-character-img" src={featuredCharacter.image} alt="featured character" />
      <div className="featured-character-info">
        <h2 className="featured-character-text">Name: {featuredCharacter.name}</h2>
        <p className="featured-character-text">Full Name: {featuredCharacter.full_name}</p>
        <p className="featured-character-text">Overview: {featuredCharacter.overview}</p>
      </div>
    </article>
  )
}

export const mapStateToProps = (store) => ({
  allCharacters: store.allCharacters,
});

FeaturedCharacter.propTypes = {
  allCharacters: PropTypes.func.isRequired
}

export default connect(mapStateToProps)(FeaturedCharacter);