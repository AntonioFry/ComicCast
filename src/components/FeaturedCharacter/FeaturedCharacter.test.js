import React from 'react';
import { FeaturedCharacter } from './FeaturedCharacter';
import { shallow } from 'enzyme';
import { mapStateToProps } from './FeaturedCharacter';

describe('FeaturedCharacter', () => {

  describe('FeaturedCharacter component', () => {
    let wrapper;
    let props;
  
    beforeEach(() => {
      props = {
        allCharacters: [
          { name: 'person', full_name: 'person person', overview: 'yeah' },
          { name: 'person', full_name: 'person person', overview: 'yeah' },
          { name: 'person', full_name: 'person person', overview: 'yeah' }
        ]
      };
      wrapper = shallow(<FeaturedCharacter allCharacters={props.allCharacters} />)
    })
  
    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('mapStateToProps', () => {

    it('should return an object with all characters', () => {
      const mockStore = {
        allCharacters: [{}, {}, {}, {}, {}],
      }

      const expected = {
        allCharacters: [{}, {}, {}, {}, {}],
      }

      const mappedProps = mapStateToProps(mockStore);

      expect(mappedProps).toEqual(expected);
    });
  });
})