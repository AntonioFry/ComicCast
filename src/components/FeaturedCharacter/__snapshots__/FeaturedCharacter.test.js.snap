// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedCharacter FeaturedCharacter component should match the snapshot 1`] = `
<article
  className="featured-character-article"
>
  <img
    alt="featured character"
    className="featured-character-img"
  />
  <div
    className="featured-character-info"
  >
    <h2
      className="featured-character-text"
    >
      Name: 
      person
    </h2>
    <p
      className="featured-character-text"
    >
      Full Name: 
      person person
    </p>
    <p
      className="featured-character-text"
    >
      Overview: 
      yeah
    </p>
  </div>
</article>
`;

exports[`FeaturedCharacter should match the snapshot 1`] = `
<article
  className="featured-character-article"
>
  <img
    alt="featured character"
    className="featured-character-img"
  />
  <div
    className="featured-character-info"
  >
    <h2
      className="featured-character-text"
    >
      Name: 
      person
    </h2>
    <p
      className="featured-character-text"
    >
      Full Name: 
      person person
    </p>
    <p
      className="featured-character-text"
    >
      Overview: 
      yeah
    </p>
  </div>
</article>
`;
